app/dashboard/views.py

```diff
@@ -3442,7 +3442,137 @@ def hackathon_onboard(request, hackathon=''):
         'referer': referer,
         'is_registered': is_registered,
     }
-    return TemplateResponse(request, 'dashboard/hackathon_onboard.html', params)
+    return TemplateResponse(request, 'dashboard/hackathon/onboard.html', params)
+
+
+def hackathon_projects(request, hackathon=''):
+    q = request.GET.get('q', '')
+    order_by = request.GET.get('order_by', '-created_on')
+    filters = request.GET.get('filters', '')
+    print(q)
+
+    page = request.GET.get('page', 1)
+    try:
+        hackathon_event = HackathonEvent.objects.filter(slug__iexact=hackathon).latest('id')
+        profile = request.user.profile if request.user.is_authenticated and hasattr(request.user, 'profile') else None
+        is_registered = HackathonRegistration.objects.filter(registrant=profile, hackathon=hackathon_event) if profile else None
+    except HackathonEvent.DoesNotExist:
+        hackathon_event = HackathonEvent.objects.last()
+
+    projects = HackathonProject.objects.filter(hackathon=hackathon_event).exclude(status='invalid').prefetch_related('profiles').order_by(order_by)
+
+    if q:
+        projects = projects.filter(
+            Q(name__icontains=q) |
+            Q(summary__icontains=q) |
+            Q(profiles__handle__icontains=q)
+        )
+    if filters == 'winners':
+        projects = projects.filter(
+            Q(badge__isnull=False)
+        )
+
+    projects_paginator = Paginator(projects, 3)
+
+    try:
+        projects_paginated = projects_paginator.page(page)
+    except PageNotAnInteger:
+        projects_paginated = projects_paginator.page(1)
+    except EmptyPage:
+        projects_paginated = projects_paginator.page(projects_paginator.num_pages)
+
+    params = {
+        'active': 'hackathon_onboard',
+        'title': 'Hackathon Projects',
+        'hackathon': hackathon_event,
+        'is_registered': is_registered,
+        'projects': projects_paginated,
+        'order_by': order_by,
+        'filters': filters,
+        'query': q.split
+    }
+
+    return TemplateResponse(request, 'dashboard/hackathon/projects.html', params)
+
+
+@csrf_exempt
+def hackathon_get_project(request, bounty_id, project_id=None):
+
+    profile = request.user.profile if request.user.is_authenticated and hasattr(request.user, 'profile') else None
+    try:
+        bounty = Bounty.objects.get(id=bounty_id)
```

Comment:
@danlipert @thelostone-mc is this a problem here with the "current" bounty? 
If the bounty is duplicated I will lost the reference to the current one? 
Im adding current() to the query but not sure this will solve that (if there is something to solve)

Reply:
answering my self: Yes is a problem, changing the query.